components/dialog/index.css

```diff
@@ -11,377 +11,420 @@ governing permissions and limitations under the License.
+	--spectrum-standard-dialog-description-font-size: var(--spectrum-standard-dialog-body-font-size);
+	--spectrum-standard-dialog-description-font-style: var(--spectrum-body-sans-serif-font-style);
+	--spectrum-standard-dialog-description-font-family: var(--spectrum-body-sans-serif-font-family);
+	--spectrum-standard-dialog-header-content-font-size: var(--spectrum-standard-dialog-body-font-size);
+
+	--spectrum-standard-dialog-heading-line-height: var(--spectrum-title-line-height);
+	--spectrum-standard-dialog-description-line-height: var(--spectrum-line-height-200);
+
+	/* Colors */
+	--spectrum-standard-dialog-heading-text-color: var(--spectrum-heading-color);
+	--spectrum-standard-dialog-description-text-color: var(--spectrum-body-color);
+	--spectrum-standard-dialog-background-color: var(--spectrum-background-layer-2-color);
+
+	/* Spacing */
+	--spectrum-standard-dialog-spacing-edge-to-content: var(--spectrum-spacing-500);
+	--spectrum-standard-dialog-spacing-title-to-description: var(--spectrum-spacing-300);
+	--spectrum-standard-dialog-spacing-description-to-footer: var(--spectrum-spacing-500);
+	--spectrum-standard-dialog-spacing-title-to-header-content: var(--spectrum-spacing-400);
+	--spectrum-standard-dialog-spacing-edge-to-header-content-inline-end: var(--spectrum-spacing-400);
+	--spectrum-standard-dialog-spacing-footer-to-button-group: var(--spectrum-spacing-400);
+
+	/* Grid */
+	--spectrum-standard-dialog-spacing-grid-padding: var(--spectrum-standard-dialog-spacing-edge-to-content);
+	--spectrum-standard-dialog-spacing-edge-to-close-button: var(--spectrum-spacing-200);
+
+	/* Hero */
+	--spectrum-standard-dialog-hero-block-size: 140px;
+
+	/* Passthrough for nested component(s) */
+	--mod-buttongroup-justify-content: flex-end;
+	--mod-buttongroup-flex-wrap: nowrap;
 }
 
 .spectrum-Dialog {
-  /* Be a flexbox to allow a full sized content area that scrolls */
-  display: flex;
-
-  /* Allow 100% width, taking into account padding */
-  box-sizing: border-box;
-
-  /* Be no bigger than max-width,  but also be 90% if the viewport is smaller than max-width */
-  inline-size: fit-content;
-  min-inline-size: var(--mod-dialog-min-inline-size, var(--spectrum-dialog-min-inline-size));
-  max-inline-size: 100%;
-
-  max-block-size: inherit;
-
-  outline: none;
+	/* Be a flexbox to allow a full sized content area that scrolls */
+	display: flex;
+
+	/* Allow 100% width, taking into account padding */
+	box-sizing: border-box;
+
+	/* Be no bigger than max-width,  but also be 90% if the viewport is smaller than max-width */
+	inline-size: var(--mod-standard-dialog-max-width, var(--spectrum-standard-dialog-max-width));
+	min-inline-size: var(--mod-standard-dialog-min-inline-size, var(--spectrum-standard-dialog-min-inline-size));
+	max-inline-size: 100%;
+	max-block-size: inherit;
+	background-color: var(--spectrum-standard-dialog-background-color);
```

Comment:
Previously the background color was being set on Modal that wraps the Dialog, so now that we're setting the background color here on the Dialog, there is some antialiasing bleed through happening on the corners since both now have background colors on top of each other:
![Screenshot 2024-11-07 at 3 22 52 PM](https://github.com/user-attachments/assets/11644621-3425-4c5f-ad1c-b5840c07c3c9)
If you set `--mod-modal-background-color: hotpink;` you can see the issue better:
![Screenshot 2024-11-07 at 3 31 34 PM](https://github.com/user-attachments/assets/ba30edf2-4d4e-424a-b73d-90f342212067)
It's kind of a CSS rendering quirk, even though these things have the same border-radius; I encountered something similar when working on color slider.

I'm not sure why Modal has the background (or why Modal is separate from Dialog to begin with), but it seems like we'll need to work toward making its background transparent? That would affect alert dialog and tray too, so I'm not sure about the best course. A stop-gap here might be to use `--mod-modal-background-color: transparent;` in this dialog template, but we can't really build it into Dialog's CSS since Modal is the parent. This could be something to talk about at team meeting.

Also, do we need a mod for this?

Reply:
I have the mod for the background here: https://github.com/adobe/spectrum-css/pull/2860/commits/c98edd36bc233651aa2acced34593cfd47d68bc2

I was able to recreate the issue with the background color, so I had to edit modal a bit to accept `customStyles`. Let me know what you think: https://github.com/adobe/spectrum-css/pull/2860/commits/3c34cba912a21b787331819fe30d54a8d08e5ba0

Reply:
That does the trick for now. It'd be a good idea to document that this is currently required by the implementation, so that doesn't slip through the cracks.

Also, what do you think about making a followup Jira issue for looking at changing Modal's CSS to have a transparent background for S2? That is, if all the components using it will have their own backgrounds.

Reply:
Here's the follow up card: https://jira.corp.adobe.com/browse/CSS-1057

[I took a shot at writing the documentation,](https://github.com/adobe/spectrum-css/pull/2860/commits/f1e6597a9aac2bb7a90103c4b46e36b9d20f50d5) but if there's a better way to way anything, I'm open to suggestions 😊

Reply:
Thanks for adding those. I'd adjust the docs slightly to remove the part about customStyles and props, since an implementation isn't using our Storybook. Something like this:
```
 * ## Usage with modal component
 * When a dialog component is used in tandem with a [modal](/docs/components-modal--docs), implementations should set `--mod-modal-background-color` to `transparent`. This will prevent any background color used in the modal from peeking through from behind the dialog at the rounded corners, allowing the dialog's background color to take precedence.
 ```

Reply:
Updated: https://github.com/adobe/spectrum-css/pull/2860/commits/70de6c8b781850573345baa0fc6b951154591e60